source/slang/slang-ast-decl.h

```diff
@@ -371,6 +371,10 @@ class FunctionDeclBase : public CallableDecl
 class ConstructorDecl : public FunctionDeclBase
 {
     SLANG_AST_CLASS(ConstructorDecl)
+   
+    // Indicates whether the declaration was synthesized by
+    // slang and not actually provided by the user
+    bool isSynthesized = false;
```

Comment:
I am still little skeptical about the word "synthesize".
It is fine as it is but can you tell me how you would define the word in this context?

Reply:
Synthesize here means a declaration generated by slang which was never actually parsed from the file.